File: verification/verify_ui.py
from playwright.sync_api import sync_playwright, expect

def verify_ui():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(viewport={'width': 1280, 'height': 720})

        # 1. Dashboard
        print("Verifying Dashboard...")
        try:
            page = context.new_page()
            page.goto("http://localhost:9002/")
            page.wait_for_load_state("networkidle")
            # Take screenshot
            page.screenshot(path="verification/dashboard.png")
            print("Dashboard screenshot taken.")
        except Exception as e:
            print(f"Dashboard verification failed: {e}")

        # 2. Playground
        print("Verifying Playground...")
        try:
            page = context.new_page()
            page.goto("http://localhost:9002/playground")
            page.wait_for_load_state("networkidle")
            page.screenshot(path="verification/playground.png")
            print("Playground screenshot taken.")
        except Exception as e:
            print(f"Playground verification failed: {e}")

        # 3. Logs
        print("Verifying Logs...")
        try:
            page = context.new_page()
            page.goto("http://localhost:9002/logs")
            page.wait_for_load_state("networkidle")
            page.screenshot(path="verification/logs.png")
            print("Logs screenshot taken.")
        except Exception as e:
            print(f"Logs verification failed: {e}")

        # 4. Alerts
        print("Verifying Alerts...")
        try:
            page = context.new_page()
            page.goto("http://localhost:9002/alerts")
            page.wait_for_load_state("networkidle")
            page.screenshot(path="verification/alerts.png")
            print("Alerts screenshot taken.")
        except Exception as e:
            print(f"Alerts verification failed: {e}")

        # 5. Config Validator
        print("Verifying Config Validator...")
        try:
            page = context.new_page()
            page.goto("http://localhost:9002/config-validator")
            page.wait_for_load_state("networkidle")
            page.screenshot(path="verification/config_validator.png")
            print("Config Validator screenshot taken.")
        except Exception as e:
            print(f"Config Validator verification failed: {e}")

        browser.close()

if __name__ == "__main__":
    verify_ui()

File: verification/verify_ui.js
/**
 * Copyright 2025 Author(s) of MCP Any
 * SPDX-License-Identifier: Apache-2.0
 */


/**
 * Verification script for the UI.
 * Captures screenshots of key pages (Dashboard, Services, Middleware) to verify visual integrity.
 */
const { chromium } = require('playwright');
const fs = require('fs');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  // Connect to the local dev server
  await page.goto('http://localhost:3000');

  // Verify Dashboard
  await page.waitForTimeout(1000); // Wait for animations/load
  await page.screenshot({ path: '/home/jules/verification/dashboard.png', fullPage: true });

  // Verify Services
  await page.goto('http://localhost:3000/services');
  await page.waitForTimeout(1000);
  await page.screenshot({ path: '/home/jules/verification/services.png', fullPage: true });

  // Verify Middleware
  await page.goto('http://localhost:3000/middleware');
  await page.waitForTimeout(1000);
  await page.screenshot({ path: '/home/jules/verification/middleware.png', fullPage: true });

  await browser.close();
})();
